corehq/apps/app_manager/util.py

```diff
@@ -658,3 +658,20 @@ def get_info(self):
             "latest_apk_version": self.get_latest_apk_version(),
             "latest_ccz_version": self.get_latest_app_version(),
         }
+
+
+def get_form_source_download_url(xform):
+    """Returns the download url for the form source for a submitted XForm
+    """
+    from corehq.apps.app_manager.models import Application
+    app = Application.get(xform.build_id)
+    try:
+        form = app.get_forms_by_xmlns(xform.xmlns)[0]
+    except KeyError:
+        return None
```

Comment:
Minor, but if this ever happens, you'll end up with a link with a blank href, right? Would be nice to hide the link instead (or maybe note that the source couldn't be found).

Reply:
https://github.com/dimagi/commcare-hq/pull/19998/commits/1ab80f1715a113a4c51c33a6b474954941e63364